core/src/main/java/org/apache/iceberg/BaseSnapshot.java

```diff
@@ -194,16 +298,51 @@ public String manifestListLocation() {
     return manifestListLocation;
   }
 
+  private void cacheChanges(FileIO fileIO) {
+    if (fileIO == null) {
+      throw new IllegalStateException("Cannot cache changes: file io is null");
+    }
+
+    ImmutableList.Builder<DataFile> adds = ImmutableList.builder();
+    ImmutableList.Builder<DataFile> deletes = ImmutableList.builder();
+
+    // read only manifests that were created by this snapshot
+    Iterable<ManifestFile> changedManifests = Iterables.filter(dataManifests(fileIO),
+        manifest -> Objects.equal(manifest.snapshotId(), snapshotId));
+    try (CloseableIterable<ManifestEntry<DataFile>> entries = new ManifestGroup(fileIO, changedManifests)
+        .ignoreExisting()
+        .entries()) {
+      for (ManifestEntry<DataFile> entry : entries) {
+        switch (entry.status()) {
+          case ADDED:
+            adds.add(entry.file().copy());
+            break;
+          case DELETED:
+            deletes.add(entry.file().copyWithoutStats());
+            break;
+          default:
+            throw new IllegalStateException(
+                "Unexpected entry status, not added or deleted: " + entry);
+        }
+      }
+    } catch (IOException e) {
+      throw new RuntimeIOException(e, "Failed to close entries while caching changes");
+    }
+
+    this.cachedAdds = adds.build();
+    this.cachedDeletes = deletes.build();
+  }
+
   private void cacheChanges() {
     if (io == null) {
-      throw new IllegalStateException("Cannot cache changes: FileIO is null");
+      throw new IllegalStateException("Cannot cache changes: file io is null");
```

Comment:
Do we need to change this? It doesn't really seem worth it.

Reply:
Nah it doesn't. Agreed.